examples/scales/asinh_demo.py

```diff
@@ -0,0 +1,58 @@
+"""
+============
+Asinh Demo
+============
```

Comment:
It would be nice to have the equation in here and the asymptotes explained.  Also to link `symlog` and argue for why this may be preferable.